### ISSUE TITLE
use of strstr() in source/compiler/prutils.c without NULL return checks 

### ISSUE THREAD

**Issue by ColinIanKing** [acpica/acpica#896]
strstr() can potentially return NULL which should be checked before accessing the return to avoid any unexpected segmentation faults to due a null pointer dereference. There are a few calls to strstr() without null checks in source/compiler/prutils.c that require checks for code safety sake:

In PrReplaceResizeSubstring():

```
        PrevOffset = Args->Offset[i];
        temp = strstr (AslGbl_MacroTokenBuffer, Args->Name);

ResetHere1:
```

and:

```
        PrevOffset = Args->Offset[i];
        temp = strstr (AslGbl_MacroTokenBuffer, Args->Name);

ResetHere2:
```

and:

```
        PrevOffset = Args->Offset[i];
        temp = strstr (AslGbl_MacroTokenBuffer, Args->Name);

ResetHere3:
        temp = strstr (temp, Args->Name);
```

Introduced by commit:
```

commit 0d7493a0b03833c78ed7b02315bf628a5236d2d3
Author: Robert Moore <Robert.Moore@intel.com>
Date:   Wed Mar 22 12:35:44 2023 -0700

    Add full support for ASL macros (preprocessor).
    The only things not supported are the tokenizing operator and
    the stringizing operator.
    
    Saket Dumbre.
```

**Comment by sacdintel** [comment]
Thank you for the code review and also all your other contributions (PRs, issues etc), sir. This is a great suggestion and being a recent grad and a new hire at Intel, I aim to only keep making better and better contributions to ACPICA in the future. I will make some changes soon and add some guardrails to the strstr() usages, but I think I also considered logical paths that can follow this route and only when a macro token matches a substring in the expansion string does it invoke the PrReplaceResizeSubstring() function meaning that the arguments to the strstr() shouldn't logically return null due to match failure. However, segmentation faults are always something to be careful about and to be avoided at all costs with some unpredictability risk in strstr() implementations (possible bugs) in GCC or Visual C or other C compilers. In the meantime, if you have good suggestions on how to resolve this issue, I would greatly appreciate that! Thanks a lot!

**Comment by sacdintel** [comment]
Fixed this issue with the latest commit as of 10/19/2023 but left the if predicate strchr() functions as they are since it is by design and also any logical program flow like I mentioned above prevents either the strchr() or strstr() functions from returning NULL since the args are already validated before calling the PrReplaceResizeSubstring() function. Thanks for reporting Colin!

**Comment by sacdintel** [comment]
Fix #896
